test/PhpSessionPersistenceTest.php

```diff
@@ -585,4 +616,111 @@ public function testStartSessionDoesNotOverrideRequiredSettings()
         $this->assertTrue($session_use_only_cookies);
         $this->assertSame('', $session_cache_limiter);
     }
+
+    public function testNoMultipleEmptySessionFilesAreCreatedIfNoSessionCookiePresent()
+    {
+        $sessionName     = 'NOSESSIONCOOKIESESSID';
+        $sessionSavePath = __DIR__ . "/sess";
+
+        $ini = $this->applyCustomSessionOptions([
+            'name'      => $sessionName,
+            'save_path' => $sessionSavePath,
+        ]);
+
+        // create a temp session save path
+        if (! is_dir($sessionSavePath)) {
+            mkdir($sessionSavePath);
+        }
+
+        // remove old session test files if any
+        $files = glob("{$sessionSavePath}/sess_*");
+        if ($files) {
+            foreach ($files as $file) {
+                unlink($file);
+            }
+        }
+
+        $persistence = new PhpSessionPersistence();
+
+        // initial sessioncookie-less request
+        $request = new ServerRequest();
+
+        for ($i = 0; $i < 3; $i += 1) {
+            $session  = $persistence->initializeSessionFromRequest($request);
+            $response = $persistence->persistSession($session, new Response());
+
+            // new request: start w/o session cookie
+            $request = new ServerRequest();
+
+            // Add the latest response session cookie value to the new request, if any
+            $setCookies = SetCookies::fromResponse($response);
+            if ($setCookies->has($sessionName)) {
+                $setCookie = $setCookies->get($sessionName);
+            }
+            if (isset($setCookie)) {
+                $cookie = new Cookie($sessionName, $setCookie->getValue());
+                $request = FigRequestCookies::set($request, $cookie);
+            }
+        }
+
+        $files = glob("{$sessionSavePath}/sess_*");
+
+        $this->assertSame(0, count($files));
```

Comment:
should can use `$this->assertCount()`: 

```suggestion
        $this->assertCount(0, $files);
```

Reply:
Thanks @samsonasik! Will add after https://github.com/zendframework/zend-expressive-session-ext/pull/38 is pulled. Otherwise phpunit failures could randomly happen. regards!

Reply:
@pine3ree #38 is in both master and develop now.

Reply:
@samsonasik using assertCount [here](https://github.com/zendframework/zend-expressive-session-ext/pull/36/commits/9de95908010b202cab006203713b768b394eb141). kind regards.